docs/src/prototypes/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react'
+import { Popup, Button, Text, Portal, Ref } from '@stardust-ui/react/src'
+import * as copyToClipboard from 'copy-to-clipboard'
+
+export type CopyToClipboardProps = {
+  value: string
+  copyPrompt: string
+  timeout?: number
+  attached?: boolean
+  pointing?: boolean
+}
+
+class CopyToClipboard extends React.Component<CopyToClipboardProps> {
+  state = {
+    copied: false,
+    copiedHeight: 0,
+    copiedWidth: 0,
+  }
+  timeoutId = undefined
+  copiedTextRef = React.createRef<HTMLElement>()
+
+  static defaultProps = {
+    timeout: 4000,
+    attached: false,
+    pointing: false,
+  }
+
+  handleClick = () => {
+    if (this.timeoutId !== undefined) {
+      clearTimeout(this.timeoutId)
+    }
+    const { timeout, value } = this.props
+    this.setState({
+      copied: true,
+    })
+    this.timeoutId = setTimeout(() => {
+      this.setState({ copied: false })
+    }, timeout)
+    copyToClipboard(value)
+  }
+
+  componentDidMount() {
+    if (this.copiedTextRef.current === null) {
+      return
+    }
+    this.setState({
+      copiedHeight: this.copiedTextRef.current.clientHeight,
+      copiedWidth: this.copiedTextRef.current.clientWidth,
+    })
+  }
+
+  render() {
+    const { attached, pointing, copyPrompt } = this.props
+    const hideContent = this.state.copied && !attached
+    const copiedStr = 'Copied to clipboard'
+    const copiedText = hideContent ? (
+      <Text size="larger" styles={{ padding: '5px 5px 5px 5px' }}>
+        {copiedStr}
+      </Text>
+    ) : (
+      <Text>{copiedStr}</Text>
+    )
+    const copyText = <Text>{copyPrompt}</Text>
+    let popupContent = this.state.copied ? copiedText : copyText
+    if (hideContent) {
+      popupContent = <Text />
+    }
+    const button = (
+      <Button
+        icon="clipboard-copied-to"
+        iconOnly
+        onClick={this.handleClick}
+        variables={siteVariables => ({
+          colorHover: !this.state.copied && siteVariables.colors.brand[400],
+        })}
+      />
```

Comment:
Can we use shorthand factory instead?

```tsx
CopyToClipboard.defaultProps = {
  button: { icon: 'clipboard-copied-to' },
}

// ...

{Button.create(button, {
  defaultProps: { iconOnly: true },
  overrideProps: () => ({ onClick }) // some work
})
```

Reply:
would we advise developers to use the factory? given that this is not internal Stardust code, but a prototype?

Reply:
Factories are public API (_but never advertised_) and exist on almost each component, we will also fix typings in #1865.